File: backup/ResCapacity.ts
import { ArrayList } from './type'

export class ResCapacity {
  public FResourceID: number // Resource ID
  public FProductID: number // Product ID
  public FProcessID: number // Process number in the product process model
  public FEfficiency: number // Efficiency
  public FCapacity1: number // Capacity 10 seconds/PCS
  public FCapacity2: number
  public FPreSetoutTime: number // Pre-setup time
  public FPostSetoutTime: number // Post-setup time
  public FProcessPassRate: number // Pass rate
  public viceResIDList: ArrayList<number> // Store副资源引用
  public FPriority: number // Main resource priority
  public FTechCharacter1: string = '' // Product current process technology feature 1
  public FTechCharacter2: string = '' // Product current process technology feature 2
  public FTechCharacter3: string = '' // Product current process technology feature 3
  public FTechCharacter4: string = '' // Product current process technology feature 4
  public FTechCharacter5: string = '' // Product current process technology feature 5
  public FTechCharacter6: string = '' // Product current process technology feature 6
  public FWorkType: number = 0 // Production method  0 Single piece production / 1 Batch production
  public FBatchWorkTime: number = 0 // Production time per batch
  public FBatchQty: number = 0 // Production quantity per batch
  public FBatchInterTime: number = 0 // Interval time between batches
  private iPreTaskID: number = 0 // Previous task ID, used to get pre-setup time
  public get IPreTaskID(): number {
    return this.iPreTaskID
  }
  public set IPreTaskID(value: number) {
    this.iPreTaskID = value
  }
  private iPreTaskTime: number = 0 // Interval time with previous task, used to determine whether to consider pre-setup time
  public get IPreTaskTime(): number {
    return this.iPreTaskTime
  }
  public set IPreTaskTime(value: number) {
    this.iPreTaskTime = value
  }
  private iTaskID: number = 0 // Task ID
  public get ITaskID(): number {
    return this.iTaskID
  }
  public set ITaskID(value: number) {
    this.iTaskID = value
  }
  private planQty: number = 0 // Planned production quantity
  public get PlanQty(): number {
    return this.planQty
  }
  public set PlanQty(value: number) {
    this.planQty = value
    if (this.FWorkType === 0) {
      // Single piece production
      this.workTime = this.planQty * this.FCapacity1
    } else if (this.FWorkType === 1) {
      // Batch production
      // Logic for batch production (not implemented in the original code)
    } else {
      this.workTime = 1
    } 
  }
  private workTime: number = 0 // Planned processing time (excluding pre and post setup time)
  public get WorkTime(): number {
    return this.workTime
  }
  public set WorkTime(value: number) {
    this.workTime = value
  }
}

File: schClass/ResCapacity.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Collections;

namespace Algorithm
{
    public class ResCapacity
    {
        public int FResourceID;     // 资源ＩＤ

        public int FProductID;// 产品ＩＤ

        public int FProcessID;// 产品工艺模型中的工序号

        public float FEfficiency;   //效率

        public float FCapacity1;   //产能 10 秒/PCS 

        public float FCapacity2;

        public float FPreSetoutTime;  //前置准备时间

        public float FPostSetoutTime; //后置准备时间

        public float FProcessPassRate;  //合格率

        public ArrayList viceResIDList; //存储副资源引用

        public int FPriority;           //主资源优先级

        public string FTechCharacter1 = "";    //产品当前工艺 工艺特征1
        public string FTechCharacter2 = "";    //产品当前工艺 工艺特征2
        public string FTechCharacter3 = "";    //产品当前工艺 工艺特征3
        public string FTechCharacter4 = "";    //产品当前工艺 工艺特征4
        public string FTechCharacter5 = "";    //产品当前工艺 工艺特征5
        public string FTechCharacter6 = "";    //产品当前工艺 工艺特征6


        public int FWorkType = 0;       //生产方式  0 单件生产 / 1 按批生产
        
        public int FBatchWorkTime = 0;       //每批生产时间

        public int FBatchQty = 0;            //每批生产数量

        public int FBatchInterTime = 0;       //每批间隔时间

        private int iPreTaskID = 0;        //前任务编号,用于取前置准备时间

        public int IPreTaskID
        {
            get { return iPreTaskID; }
            set { iPreTaskID = value; }
        }

        private int iPreTaskTime = 0;      //与前任务间隔时间,用于是否考虑前置准备时间

        public int IPreTaskTime
        {
            get { return iPreTaskTime; }
            set { iPreTaskTime = value; }
        }

        private int iTaskID = 0;         //任务编号

        public int ITaskID
        {
            get { return iTaskID; }
            set { iTaskID = value; }
        }

        private float planQty = 0;      //计划生产数量

        public float PlanQty
        {
            get { return planQty; }
            set { planQty = value; 

                  //计算加工时间
                  if ( FWorkType == 0 )    //单件生产
                  {
                      workTime = planQty * FCapacity1 ;   
                  }
                  else if ( FWorkType == 1 )  //按批生产
                  {
                     // workTime = planQty / ;
                  }
                  else
                  {                      
                      workTime =    1;
                  }
            }
        }

        private float workTime = 0;      //计划加工时间（不包括前、后准备时间）

        public float WorkTime
        {
            get { return workTime; }
            set { workTime = value; }
        }

       
    }


}
